packages/flutter/test/rendering/editable_test.dart

```diff
@@ -783,15 +732,15 @@ void main() {
     editable.selection = const TextSelection.collapsed(offset: 0);
     pumpFrame();
 
-    await simulateKeyDownEvent(LogicalKeyboardKey.arrowRight, platform: 'android');
-    await simulateKeyUpEvent(LogicalKeyboardKey.arrowRight, platform: 'android');
+    editable.moveSelectionRight(SelectionChangedCause.keyboard);
```

Comment:
I don't quite understand why we need to change this line? As I understand it we should be testing from the keyboard event instead of calling editable.moveSelectionRightFoo directly.

Reply:
RenderEditable no longer has this keyboard handling, so if I use simulateKeyDownEvent etc., then it won't be affected.

At a higher level, Shortcuts will receive the keyboard event, which will call TextEditingActions, which will call RenderEditable's moveSelectionRight.  In editable_text_test.dart there are tests that cover keyboard events causing changes in selection at a high level like this.  Here, I think it's only necessary to test these methods directly.

If I was writing these tests from scratch I probably would organize them differently, though...

Reply:
Oh, right, sorry I misunderstood.